expand/environ.go

```diff
@@ -161,7 +162,19 @@ func listEnvironWithUpper(upper bool, pairs ...string) Environ {
 			}
 		}
 	}
-	sort.Strings(list)
+
+	sort.SliceStable(list, func(i, j int) bool {
+		isep := strings.IndexByte(list[i], '=')
+		jsep := strings.IndexByte(list[j], '=')
+		if isep < 0 {
+			isep = 0
+		}
+		if jsep < 0 {
+			jsep = 0
+		}
+		return strings.Compare(list[i][0:isep], list[j][0:jsep]) < 0
```

Comment:
just use `<`?

Reply:
you can omit the `0:`.

Reply:
Dropped the zero but not sure what you mean by "just use <" ?

Reply:
No need for strings.Compare, it's there just for completeness. You can do str1 < str2.

Reply:
Oh, of course. :)